lib/graphiti/resource/dsl.rb

```diff
@@ -9,6 +9,12 @@ def filter(name, *args, &blk)
           type_override = args[0]
 
           if (att = get_attr(name, :filterable, raise_error: :only_unsupported))
+            type = type_override || att[:type]
+
+            if type == :enum && opts[:allow].blank?
```

Comment:
Look in the filter code for `canonical` and go that way. The issue here is you might create a type off of enum, like `foo_enum` - you want it to go through all the same enum logic, but with this new type name. That's why types have a `canonical_name`.

Reply:
If you had a `string_enum` and `integer_enum` type, would you expect them to both have a `canonical_name == :enum`?

Reply:
Yes, exactly

Reply:
That is why, even if we don't do anything with integers in this PR, we should check the canonical name